src/SprykerEco/Zed/CrefoPay/Persistence/CrefoPayRepository.php

```diff
@@ -0,0 +1,212 @@
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        if ($entityTransfer === null) {
+            return new PaymentCrefoPayTransfer();
+        }
+
+        return $this->getFactory()
+            ->createCrefoPayPersistenceMapper()
+            ->mapEntityTransferToPaymentCrefoPayTransfer($entityTransfer, new PaymentCrefoPayTransfer());
+    }
+
+    /**
+     * @param int $idSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByIdSalesOrder(int $idSalesOrder): PaymentCrefoPayTransfer
+    {
+        $query = $this->getPaymentCrefoPayQuery()->filterByFkSalesOrder($idSalesOrder);
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        if ($entityTransfer === null) {
+            return new PaymentCrefoPayTransfer();
+        }
+
+        return $this->getFactory()
+            ->createCrefoPayPersistenceMapper()
+            ->mapEntityTransferToPaymentCrefoPayTransfer($entityTransfer, new PaymentCrefoPayTransfer());
+    }
+
+    /**
+     * @param string $pspReference
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByPspReference(string $pspReference): PaymentCrefoPayTransfer
+    {
+        $query = $this->getPaymentCrefoPayQuery()->filterByPspReference($pspReference);
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        if ($entityTransfer === null) {
+            return new PaymentCrefoPayTransfer();
+        }
+
+        return $this->getFactory()
+            ->createCrefoPayPersistenceMapper()
+            ->mapEntityTransferToPaymentCrefoPayTransfer($entityTransfer, new PaymentCrefoPayTransfer());
+    }
+
+    /**
+     * @param int $idSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayOrderItemTransfer[]
+     */
+    public function getAllPaymentCrefoPayOrderItemsByIdSalesOrder(int $idSalesOrder): array
+    {
+        $query = $this->getPaymentCrefoPayOrderItemQuery()
+            ->useSpyPaymentCrefoPayQuery()
+                ->filterByFkSalesOrder($idSalesOrder)
+            ->endUse();
+
+        $entityTransfers = $this->buildQueryFromCriteria($query)->find();
+
+        $mapper = $this->getFactory()->createCrefoPayPersistenceMapper();
+        $result = [];
+
+        foreach ($entityTransfers as $entityTransfer) {
+            $result[] = $mapper
+                ->mapEntityTransferToPaymentCrefoPayOrderItemTransfer(
+                    $entityTransfer,
+                    new PaymentCrefoPayOrderItemTransfer()
+                );
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param int[] $orderItemIds
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayOrderItemTransfer[]
+     */
+    public function getOrderItemsByIdsSalesOrderItems(array $orderItemIds): array
```

Comment:
according to agreement in core, this (and other similar methods) should be called `findXXxx`